test/pom.xml

```
@@ -205,6 +205,13 @@ THE SOFTWARE.
         <!-- Version specified in grandparent POM -->
         <extensions>true</extensions>
         <executions>
+          <execution>
+           <id>prepare-test-plugins</id>
+           <goals>
+             <goal>resolve-test-dependencies</goal>
+           </goals>
+           <phase>generate-test-resources</phase>
+          </execution>
```

@jtnord Have you looked at the CI build results since your commits 4 days ago? You have a linter issue. Also, in the plugin lifecycle defined in `maven-hpi-plugin` the `resolve-test-dependencies` goal runs as part of the `test-compile` phase:

https://github.com/jenkinsci/maven-hpi-plugin/blob/b3d4d40c57eb956fe58153d7b350aa7d2d673090/src/main/resources/META-INF/plexus/components.xml#L21=

Since the objective of the `test/` module in core is very similar to that of a plugin's test suite, I think we should keep these two parallel implementations as similar as possible by using the same phase here:

```suggestion
          <execution>
            <id>prepare-test-plugins</id>
            <goals>
              <goal>resolve-test-dependencies</goal>
            </goals>
            <phase>test-compile</phase>
          </execution>
```

---

> @jtnord Have you looked at the CI build results since your commits 4 days ago? 

I took a quick look.  What I saw was a mess of logspam and connection disconnects.

Generally my mode of working is to address the first discovered issue and as jenkins war was spewing errors during the build (which is before Jenkins test) I fixed those first - which turned out to not be issues.

if the build is failing due to connection attempts I do not even bother spending any effort, especially for something that is not actually an production bug or blocking anything

> You have a linter issue. 

Thanks, I had missed that in the noise.

> Also, in the plugin lifecycle defined in `maven-hpi-plugin` the `resolve-test-dependencies` goal runs as part of the `test-compile` phase:
> 
> https://github.com/jenkinsci/maven-hpi-plugin/blob/b3d4d40c57eb956fe58153d7b350aa7d2d673090/src/main/resources/META-INF/plexus/components.xml#L21=
> 
> Since the objective of the `test/` module in core is very similar to that of a plugin's test suite, I think we should keep these two parallel implementations as similar as possible by using the same phase here:

🤔   I agree they should be the same - What I do not agree with is the phase that the `hpi` plugin uses, we are generating a resource not compiling so the `hpi` plugin seems incorrect here.  Taking a quick look there does not seem to be any need for that ordering in the `hpi-plugin` and an tooling may expect to be able to run the phases up to but excluding `test-compile` and then compile with its own internal compiler without issues.  I guess this is one reason why eclipse often struggles and you need to drop the the CLI to invoke maven.

That said the HPI plugin and packaging is a mess of how not to do things with maven so I am not going to die on this hill or be responsible for any fallout of the change.